storage/src/cache/cachedfile.rs

```diff
@@ -579,7 +579,30 @@ impl FileCacheEntry {
         }
 
         if !bitmap.wait_for_range_ready(chunk_index, count)? {
-            Err(eio!("failed to read data from storage backend"))
+            if !prefetch {
```

Comment:
Minor suggestion:
```
if prefetch {
 return Err(eio!("failed to read data from storage backend"));
}
...
```
This function is a little over complex, will try to refactor it later

Reply:
Thanks , updated. 
Yes , this func is too long now , and I remember seems there's still issue for stargz part, may need farther check later.